### PR TITLE
Implement Hernando Cortez

### PR DESCRIPTION
**Pull request by nathanj**
I tested with a few of the ICE listed in the FAQ for this card. If you think there are any cards that this does not work with let me know, and I will add a test for it.

### CODE REVIEW

**Comment by nathanj** [comment]
Yeah that makes sense. I was worried about the runtime performance since it is dealing with strings, but now that I think about it this is not going to hit that often.

**Comment by nathanj** [comment]
Made the change. I ran into a small problem because `:text` was removed from the card when it is placed in the hand due to `make-card`. I'm not sure why it was removed. I added it back.

Another option would be to get the card from `@all-cards` when Hernando fires if you want to keep the removal of `:text`.

**Comment by NoahTheDuke** [comment]
I think the goal for `dissoc`ing the card text was to make the card objects less weighty, as we don't ever modify the card text and it can be easily retrieved by calling `(-> card :title server-card :text)`.

File: src/clj/game/cards/resources.clj [hunk 1]
```diff
@@ -48,6 +48,15 @@
     ;; Add an effect only if there is none in the map
     (merge trash-effect definition)))
 
+(defn count-num-subroutines
+  "Count the number of subroutines for Hernando Cortez. Note that this only
+  counts subroutines that exist as part of the card text. It does not count
+  subroutines that the card will gain once rezzed."
+  [card]
+  (->> (split-lines (:text card))
```
Comment (NoahTheDuke) [inline src/clj/game/cards/resources.clj]:
```suggestion
  (->> (-> card :title server-card :text split-lines)
```

File: src/clj/game/core/turns.clj [hunk 2]
```diff
@@ -128,7 +128,7 @@
   ([card cid]
   (-> card
       (assoc :cid cid :implementation (card-implemented card))
-      (dissoc :setname :text :_id :influence :number :influencelimit :factioncost))))
+      (dissoc :setname :_id :influence :number :influencelimit :factioncost))))
```
Comment (NoahTheDuke) [inline src/clj/game/core/turns.clj]: Revert this change.

**Comment by NoahTheDuke** [comment]
Closes #3962
